ballerina/tests/mock_service.bal

```diff
@@ -0,0 +1,288 @@
+// Copyright (c) 2024 WSO2 LLC. (https://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+import ballerina/http;
+
+listener http:Listener httpListener = new (9090);
+
+service / on httpListener {
+
+    resource function get rest/api/'3/myself(string? expand) returns User|http:Unauthorized {
+        return {
+            "name": "Mia Krystof",
+            "active": true
+        };
+    }
+
+    // Mock response for `getMyPermissions`
+    resource function get rest/api/'3/mypermissions(string? projectKey, string? projectId, string? issueKey, string? issueId, string? permissions, string? projectUuid, string? projectConfigurationUuid, string? commentId) returns Permissions|ErrorCollectionBadRequest|ErrorCollectionUnauthorized|ErrorCollectionNotFound {
+    return{
+        permissions: {
+                "BROWSE_PROJECTS": {
+                    id: "10",
+                    key: "BROWSE_PROJECTS",
+                    name: "Browse Projects",
+                    description: "Ability to browse projects and the issues within them.",
+                    havePermission: true 
+                }
+            }
+    }
+    }
+
+
+    // Mock response for `post rest/api/'3/role`
+    resource function post rest/api/'3/role(@http:Payload CreateUpdateRoleRequestBean payload) returns ProjectRoleOk|http:BadRequest|http:Unauthorized|http:Forbidden|http:Conflict {
+    return{
+        id: 101,
+        name: payload.name,
+        description: payload.description
+    }
+    }
+
+     // New mock for `delete rest/api/'3/role/[int id]`
+     resource function delete rest/api/'3/role/[int id](int? swap) returns http:NoContent|http:BadRequest|http:Unauthorized|http:Forbidden|http:NotFound|http:Conflict {
+        http:Response response = new;
+        response.statusCode = 204;
+        return response;
```

Comment:
This probably should give you compilation errors.

```suggestion
        return http:NO_CONTENT;
```

Reply:
Have you run these tests locally? Please run the tests and see. You'll get the compilation errors.